### ISSUE TITLE
Mousing off the end of a slider drag doesn't change value

### ISSUE THREAD

**Issue by flukejones** [slint-ui/slint#5173]
With the recent slider `released()` callback, I've noticed that if the interaction dragging the slider goes off the end of the slider track then the release event doesn't fire and you have to go back and click once.

**Comment by tronical** [comment]
This patch fixes it for me (the Qt slider already implements this):

```patch
diff --git a/internal/compiler/widgets/common/slider-base.slint b/internal/compiler/widgets/common/slider-base.slint
index 2451ecf08..cbdcb6952 100644
--- a/internal/compiler/widgets/common/slider-base.slint
+++ b/internal/compiler/widgets/common/slider-base.slint
@@ -32,16 +32,15 @@ export component SliderBase {
         width: 100%;
         height: 100%;

-        clicked => {
-            released(root.value);
-        }
-
         pointer-event(event) => {
             if (event.button != PointerEventButton.left) {
                 return;
             }

             if (event.kind == PointerEventKind.up) {
+                if (root.handle-pressed) {
+                    released(root.value);
+                }
                 root.handle-pressed = false;
                 return;
             }
```

but I'm having trouble making at test for it.

I tried this, but somehow the pointer event isn't triggered:

```patch
diff --git a/tests/cases/widgets/slider_basic.slint b/tests/cases/widgets/slider_basic.slint
index e656ca84e..d91a6bd0c 100644
--- a/tests/cases/widgets/slider_basic.slint
+++ b/tests/cases/widgets/slider_basic.slint
@@ -10,12 +10,14 @@ export component TestCase inherits Window {
         minimum: 0;
         maximum: 100;
         value: 10;
+        accessible-label: "testslider";
     }

     forward-focus: slider;
     out property <bool> slider-focused <=> slider.has_focus;
     callback released <=> slider.released;
     callback changed <=> slider.changed;
+    in property value <=> slider.value;
 }

 /*
@@ -60,6 +62,28 @@ slint_testing::send_keyboard_char(&instance, 'a', false);
 assert_eq!(edits.borrow().clone(), vec![14f32]);
 assert_eq!(changes.borrow().clone(), vec![11f32, 12f32, 13f32, 14f32]);

+edits.borrow_mut().clear();
+
+// Reset to center
+instance.set_value(50.);
+
+let slider = slint_testing::ElementHandle::find_by_accessible_label(&instance, "testslider").next().unwrap();
+
+// Press center and then drag outside
+let mut position = slider.absolute_position();
+position.x += slider.size().width / 2.;
+position.y += slider.size().height / 2.;
+let button = slint::platform::PointerEventButton::Left;
+instance.window().dispatch_event(slint::platform::WindowEvent::PointerMoved { position });
+instance.window().dispatch_event(slint::platform::WindowEvent::PointerPressed { position, button });
+slint_testing::mock_elapsed_time(50);
+for x in (position.x as usize..200).step_by(10) {
+    position.x = x as _;
+    instance.window().dispatch_event(slint::platform::WindowEvent::PointerMoved { position });
+}
+instance.window().dispatch_event(slint::platform::WindowEvent::PointerReleased { position, button });
+assert_eq!(edits.borrow().clone(), vec![100f32]);
+
 ```

 */
```

@ogoffart @FloVanGH do you have any idea what I'm doing wrong?
